### ISSUE TITLE
about the loss

### ISSUE THREAD

**Issue by JoyHuYY1412** [aimerykong/Recurrent-Pixel-Embedding-for-Instance-Grouping#5]
Thank you for your work. I am not very familiar with MATLAB coding, but according to your paper, I think the loss you used for instance segmentation is:
the margin loss after every step of mean-shift from 0 to T, and each time step t you use the updated data to get the loss lt
Also, since the mean-shift kernel is set (according to your paper), all you train is the embedding from pixel to the 64-dimension feature. Am I right?

**Comment by aimerykong** [comment]
Yes. You can think in this way. The mean-shift behaves more like a loss
rather than parametric module. But here are two things to notice --
1. In the toy dataset (mnist), I set T as large as 5, but in practice, I
use T=1 because the GPU memory is not enough to hold multiple copies of the
transformed data. Of course this can be improved in different ways but I
didn't push this in this paper.
2. My remark (1) also indicates a different thing from what you say about
the mean-shift kernel. Actually, the kernel is computed in each loop based
on the new data, not the original 64-dim raw features.

Hope this helps. Thanks for your interest.

**Comment by JoyHuYY1412** [comment]
Yes, thank you  for your help. I meant that the kernel's form is fixed since you choose a fixed bandwidth, while the data is updated each mean-shift step.
